modules/stochastic_tools/python/visualize_sobol.py

```diff
@@ -0,0 +1,249 @@
+
+            if len(data['vector_names']) > 1:
+                if i > 0:
+                    tab['Value'].append(' ')
+                else:
+                    tab['Value'].append(vname)
+
+
+    return pd.DataFrame(tab)
+
+def barPlot(data, stat, log_scale):
+
+    bars = []
+    for i, pnamei in enumerate(data['param_names']):
+        for j, pnamej in enumerate(data['param_names']):
+            if j > i:
+                break;
+            if opt.stat == 'SECOND_ORDER':
+                name = pnamei + ", " + pnamej
+                ind = (i, j)
+            elif j == i:
+                name = pnamei
+                ind = i
+            else:
+                continue
+
+            y = [data[vname]['val'].item(ind) for vname in data['vector_names']]
+            error_y = None
+            if data['ci_levels'] is not None:
+                error_y = dict(type='data')
+                error_y['arrayminus'] = [data[vname]['val'].item(ind) - data[vname]['ci_minus'].item(ind) for vname in data['vector_names']]
+                error_y['array'] = [data[vname]['ci_plus'].item(ind) - data[vname]['val'].item(ind) for vname in data['vector_names']]
+            bars.append(go.Bar(name=name, x=data['vector_names'], y=y, error_y=error_y))
+
+    if opt.stat == 'TOTAL':
+        ylabel = '$S_T$'
+    elif opt.stat == 'FIRST_ORDER':
+        ylabel = '$S_i$'
+    elif opt.stat == 'SECOND_ORDER':
+        ylabel = '$S_{i,j}$'
+
+    layout=go.Layout(barmode='group', yaxis=dict(title=ylabel, type=('log' if log_scale else 'linear')))
+    return go.Figure(data=bars, layout=layout)
+
+def heatmap(data, stat, log_scale):
+
+    vnames = data['vector_names']
+    fig = plotly.subplots.make_subplots(rows=len(vnames), cols=1, subplot_titles=vnames, vertical_spacing=0.15/len(vnames))
+
+    for i, vname in enumerate(vnames):
+        row = (i // 2) + 1
+        col = (i % 2) + 1
+
+        z = data[vname]['val']
+        if stat != 'SECOND_ORDER':
+            z = np.diag(z)
+
+        if log_scale:
+            zmin = int(np.floor(np.amin(np.log10(z[z>0]))))
+            zmax = int(np.ceil(np.amax(np.log10(z[z>0]))))
+            ztick = np.linspace(zmin, zmax, zmax-zmin+1, dtype=int)
+            ztick_lab = ['10<sup>{}</sup>'.format(zt) for zt in ztick]
+            z[z<0] = np.nan
+            z = np.log10(z)
+        else:
+            ztick = None
+            ztick_lab = None
+        if stat == 'TOTAL':
+            ctitle = '$S_T$'
+        elif stat == 'FIRST_ORDER':
+            ctitle = '$S_i$'
+        elif stat == 'SECOND_ORDER':
+            ctitle = '$S_{i,j}$'
+        colorbar = go.heatmap.ColorBar(title=dict(text=ctitle, side='right'), x=1, y=(i+0.5)/len(vnames), len=1/len(vnames), tickvals=ztick, ticktext=ztick_lab)
+
+        fig.add_trace(go.Heatmap(x=data['param_names'], y=data['param_names'], z=z, colorbar=colorbar), row=i+1, col=1)
+
+    return fig
+
+if __name__ == '__main__':
```

Comment:
I would put all this in to a "main()" that returns 0 at the end, then change this to 

```
sys.exit(main())
```

This allows other tools to import the main function desired for wrapping it with other utils